tests/compile/test_full_graph.py

```diff
@@ -1,16 +1,75 @@
 import os
 
 import pytest
+import torch
 
-from vllm.utils import cuda_device_count_stateless
+from tests.quantization.utils import is_quant_method_supported
+from vllm import LLM, SamplingParams
+from vllm.compilation.backends import vllm_backend
+from vllm.plugins import set_torch_compile_backend
+from vllm.utils import cuda_device_count_stateless, is_hip
 
 from ..utils import fork_new_process_for_each_test
 
+TEST_MODELS = [
+    ("facebook/opt-125m", {}),
+    ("nm-testing/tinyllama-oneshot-w8w8-test-static-shape-change", {
+        "dtype": torch.float16,
+        "quantization": "compressed-tensors"
+    }),
+    ("neuralmagic/Meta-Llama-3-8B-Instruct-FP8", {
+        "dtype": torch.float16,
+        "quantization": "fp8"
+    }),
+    ("nm-testing/Meta-Llama-3-8B-Instruct-W8A8-Dyn-Per-Token-2048-Samples", {
+        "quantization": "compressed-tensors"
+    }),
+    ("meta-llama/Meta-Llama-3-8B", {}),
+]
 
-@pytest.mark.parametrize("model", ["meta-llama/Meta-Llama-3-8B"])
+# TODO: enable in pytorch 2.5
+if False and is_quant_method_supported("aqlm"):  # noqa: SIM223
+    TEST_MODELS.append(("ISTA-DASLab/Llama-2-7b-AQLM-2Bit-1x16-hf", {
+        "quantization": "aqlm"
+    }))
+
+# TODO: enable in pytorch 2.5
+if False and is_quant_method_supported("gguf"):  # noqa: SIM223
+    TEST_MODELS.append(("TheBloke/TinyLlama-1.1B-Chat-v1.0-GGUF", {
+        "quantization": "gguf"
+    }))
+
+if is_quant_method_supported("gptq"):
+    TEST_MODELS.append(("TheBloke/TinyLlama-1.1B-Chat-v0.3-GPTQ", {
+        "quantization": "gptq"
+    }))
+
+if is_quant_method_supported("gptq_marlin"):
+    TEST_MODELS.append(("TheBloke/TinyLlama-1.1B-Chat-v1.0-GPTQ", {
+        "quantization": "gptq_marlin"
```

Comment:
Depending on the device (i.e. H100 or A100) this will currently run either run the Machete (H100) or Marlin kernels (anything Ampere) (after https://github.com/vllm-project/vllm/commit/86e9c8df29a954a7a2fc46e9985fecc2a2e15ae8) (and hopefully gptq_marlin will be deprecated post [refactor](https://docs.google.com/document/d/1AfgGfF73H_hcXfw6ehYO_l1vHEItsopbxFoV1PvnGIQ/edit?usp=sharing), currently the only way to explicitly control would be to set `VLLM_DISABLED_KERNELS` (a bit of a hacky env var for performance sweeps) 

Im not sure what makes more sense here, maybe make `VLLM_DISABLED_KERNELS` a more "official env var" and control the env here (maybe flip the logic so its `VLLM_ENABLED_KERNELS`) or just test on more devices

Reply:
I'm ok with removing `gptq_marlin` as long as one of the other quantization methods exercises the machete kernels.

My goal with the tests is to make sure the different quantization op variants are covered by a fullgraph test.